### PR TITLE
Uber refactor

### PR DESCRIPTION
**Pull request by Ozymadias**

### CODE REVIEW

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java [hunk 1]
```diff
@@ -46,19 +48,18 @@ private Parent createContent() {
     private EventHandler<MouseEvent> getMove() {
         return event -> {
             Cell cell = (Cell) event.getSource();
-            if (whichPlayer) {
-                if (!running || cell.wasShot)
-                    return;
-                handlePlayersMove(cell);
-            }
-            if (!running)
+            if (!isMyTurn || !shipPlacer.areAllShipsPlaced() || cell.wasShot)
```
Comment (smolamarcin) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java]: Missing braces

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Ship.java [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
-        return health!=0;
+        return health != 0;
     }
 
-    int getRemainingHealth() {
-        return battleShip.getShipsRemainingHealth(battleShip.getShipProperties());
+    int getLength() {
+        return positions.size();
     }
 }
```
Comment (smolamarcin) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Ship.java]: Missing empty line

File: battleship-client/src/main/java/com/academy/solid/nie/client/communication/IpValidator.java [hunk 3]
```diff
@@ -22,7 +22,7 @@ public boolean validate(String ip) {
         if (splitted.length != 4)
             return false;
         for (String s : splitted)
-            if (!(s.matches("[1-9]\\d*") && Integer.parseInt(s) < 256))
+            if (!(s.matches("[0-9]\\d*") && Integer.parseInt(s) < 256))
```
Comment (rivalez) [inline battleship-client/src/main/java/com/academy/solid/nie/client/communication/IpValidator.java]: add regex to const

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java [hunk 4]
```diff
@@ -102,39 +101,14 @@ boolean isShipPositionValid(Ship ship, Cell cell) {
...
-     *
-     * @param ship - represents single instance of Ship
-     * @param x - the horizontal coordinate of the field on which the ship will be placed
-     * @param y - the vertical coordinate of the field on which the ship will be placed
-     */
-    private void placeShipVertically(Ship ship, int x, int y) {
-        for (int i = y; i < y + ship.getRemainingHealth(); i++) {
-            Cell cell = getCell(x, i).addShip(ship);
+    private void placeShip(Ship ship) {
+        for (Point2D point:ship.getPositions()) {
```
Comment (rivalez) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: spaces after point and before ship (formatting)

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java [hunk 5]
```diff
@@ -212,37 +186,26 @@ Cell getCell(int x, int y) {
...
-            for (int i = x; i < x + length; i++)
-                if (!isInScope(i, y) || getCell(i, y).isOccupied() || canPlaceShip(y, i))
-                    return false;
-        }
-        return true;
+        return ship.getPositions().stream().noneMatch(point2D ->
+        {
+            int x = point2D.getX();
+            int y = point2D.getY();
+            return !isInScope(x, y) || getCell(x, y).isOccupied() || !canPlaceShip(x, y);
```
Comment (rivalez) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: inverse negative conditions

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java (hunk 1 above)
Comment (rivalez) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java]: inverse negative conditions

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 6]
```diff
@@ -29,38 +28,55 @@
...
+        return event.getButton() == MouseButton.PRIMARY ? Type.VERTICAL : Type.HORIZONTAL;
+    }
+
+    private List<Point2D> produceCells(Cell cell, Type type) {
+        List<Point2D> points = new ArrayList<>();
+        Integer peek = typesOfShips.peek();
+        int x = cell.getCellX();
+        int y = cell.getCellY();
+        points.add(Point2D.of(x, y));
+        for (int length = peek - 1; length > 0; length--) {
```
Comment (smolamarcin) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: Magic number? peek -1

File: battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java [hunk 7]
```diff
@@ -46,9 +41,7 @@ public void shouldNotPlaceShip_WhenThereIsAlreadyShip() {
         EventHandler<MouseEvent> handler = Mockito.mock(EventHandler.class);
         board.initialize(handler);
 
-        ShipFactory factory = new HorizontalShipFactory();
-        BattleShip battleShip = factory.createShip(Collections.singletonList(Point2D.of(1, 6)));
-        Ship ship = new Ship(battleShip);
+        Ship ship = new Ship(Collections.singletonList(Point2D.of(1, 6)));
```
Comment (smolamarcin) [inline battleship-client/src/test/java/com/academy/solid/nie/client/ui/BoardTest.java]: Magic number

File: common/src/main/java/com/academy/solid/nie/ships/Type.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package com.academy.solid.nie.ships;
+
+public enum Type {
```
Comment (smolamarcin) [inline common/src/main/java/com/academy/solid/nie/ships/Type.java]: Missing javadoc.

**Pull request merged**